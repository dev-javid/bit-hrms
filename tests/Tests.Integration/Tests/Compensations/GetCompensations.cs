namespace Tests.Integration.Tests.Compensations
{
    public class GetCompensations(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/compensations";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data_Of_Other_Employee_For_Company_Admin()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Compensations/GetCompensations.sql");

            using HttpResponseMessage response = await Client.GetAsync($"{Route}?EmployeeId=2");

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data_Of_Own_Employee()
        {
            await LoginAsEmployeeAsync();
            await FeedDataAsync("Tests/Compensations/GetCompensations.sql");

            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
