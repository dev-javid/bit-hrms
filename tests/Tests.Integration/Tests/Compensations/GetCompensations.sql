INSERT INTO public.compensations
(id, company_id, employee_id, effective_from, amount, created_by, created_on, modified_by, modified_on)
VALUES
(1, 999, 1, '2024-10-03', decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(2, 999, 2, '2024-03-03', decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), 0, '2024-05-03 03:31:32.227954-07', NULL, NULL);
