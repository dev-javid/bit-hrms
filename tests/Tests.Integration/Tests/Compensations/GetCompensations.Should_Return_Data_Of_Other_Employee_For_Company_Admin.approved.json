[
    {
        "compensationId": 2,
        "employeeId": 2,
        "effectiveFrom": "2024-03-03",
        "amount": 1200
    }
]